nemo/collections/llm/gpt/model/base.py

```diff
@@ -48,7 +48,7 @@ def configure_model(self, tokenizer) -> "MCoreGPTModel":
 
         return MCoreGPTModel(
             self,
-            transformer_layer_spec=get_gpt_layer_with_transformer_engine_spec(),
+            transformer_layer_spec=get_gpt_layer_with_transformer_engine_spec(getattr(self, 'num_moe_experts', 1)),
```

Comment:
Do you think it makes more sense to expose this into the `GPTConfig`?

Reply:
sure